src/Stratis.Bitcoin.Features.MemoryPool/MempoolBehavior.cs

```diff
@@ -315,25 +315,22 @@ public override object Clone()
             {
                 //inv.type |= nFetchFlags;
 
-                // TODO: This is incorrect, in blocks only mode we should just add to known inventory but not relay
-                if (this.isBlocksOnlyMode)                 
-                    this.logger.LogInformation("Transaction ID '{0}' inventory sent in violation of protocol peer '{1}'.", inv.Hash, peer.RemoteSocketEndpoint);
+                lock (this.lockObject)
+                {
+                    this.filterInventoryKnown.Add(inv.Hash);
+                }
 
-                if (await this.orphans.AlreadyHaveAsync(inv.Hash))
+                if (this.isBlocksOnlyMode)
+                {
+                    this.logger.LogInformation("Transaction ID '{0}' inventory sent in violation of protocol peer '{1}'.", inv.Hash, peer.RemoteSocketEndpoint);
+                }
+                else if (await this.orphans.AlreadyHaveAsync(inv.Hash))
                 {
                     this.logger.LogDebug("Transaction ID '{0}' already in orphans, skipped.", inv.Hash);
-                    continue;
                 }
-
-                send.Inventory.Add(inv);
-            }
-
-            // add to known inventory
-            lock (this.lockObject)
-            {
-                foreach (InventoryVector inventoryVector in send.Inventory)
+                else
```

Comment:
The original version which continue seems little more readable to me, prevents you to have else if statements. Not sure if others feel the same.